uhyve-interface/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+//! # Uhyve Hypervisor Interface
+//!
+//! The uhyve hypercall interface works as follows:
+//!
+//! - On x86_64 you use a out port instruction. The address of the out_port corresponds to the
+//! hypercall you want to use. You can obtain it from the [`IoPorts`] enum. The data send to
+//! that port is the physical memory (in the VM) of the parameters of that hypercall.
+//! - On aarch64 you write to the respective [`HypercallAddress`]. The 64-bit data written to that location is the guest's physical memory address of the hypercall's parameter.
+
+#![no_std]
+
+// TODO: Throw this out, once https://github.com/rust-lang/rfcs/issues/2783 is resolved
+use num_enum::TryFromPrimitive;
+
+pub mod elf;
+pub mod parameters;
+use parameters::*;
+
+/// The version of the uhyve interface. Note: This is not the same as the semver of the crate but
+/// should be increased on every version bump.
```

Comment:
A change in the semver of the crate does not imply a change in the hypercall API. The user-facing API might change in a way that requires a crate semver bump without any change to the hypercall version.

Reply:
Better now?